src/utils.rs

```diff
@@ -199,32 +199,58 @@ pub fn projv(a: &Vector3<f64>, b: &Vector3<f64>) -> Vector3<f64> {
     b * a.dot(b) / b.dot(b)
 }
 
-/// Computes the RSS state errors in two provided vectors
+/// Computes the Root Sum Squared (RSS) state errors between two provided vectors.
+///
+/// # Arguments
+///
+/// * `prop_err` - A vector representing the propagated error.
+/// * `cur_state` - A vector representing the current state.
+///
+/// # Returns
+///
+/// * A f64 value representing the RSS state error.
 pub fn rss_errors<N: DimName>(prop_err: &OVector<f64, N>, cur_state: &OVector<f64, N>) -> f64
 where
     DefaultAllocator: Allocator<f64, N>,
 {
-    let mut v = 0.0;
-    for i in 0..N::dim() {
-        v += (prop_err[i] - cur_state[i]).powi(2);
-    }
-    v.sqrt()
+    prop_err
```

Comment:
Good idea! I initially seem to have written this in April 2018 (the 2021 commit is got a change in nalgebra), and that was before I used any FP in Rust.

Reply:
I added the UT based on the way you created OVector in the code snippet you shared.